dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -483,6 +483,27 @@ public Response setAggregatedMetrics(String id, String versionId, Partner platfo
         return Response.ok().entity(version.getMetricsByPlatform()).build();
     }
 
+    @Override
+    public Response getAggregatedMetrics(String id, String versionId, Partner platform) {
+        Entry<?, ?> entry;
+        try {
+            entry = getEntry(id, Optional.empty());
+        } catch (UnsupportedEncodingException | IllegalArgumentException e) {
+            return BAD_DECODE_REGISTRY_RESPONSE;
+        }
+
+        if (entry == null) {
+            throw new CustomWebApplicationException(TOOL_NOT_FOUND_ERROR, HttpStatus.SC_NOT_FOUND);
+        }
+
+        Version<?> version = getVersion(entry, versionId).orElse(null);
+        if (version == null) {
+            throw new CustomWebApplicationException(VERSION_NOT_FOUND_ERROR, HttpStatus.SC_NOT_FOUND);
+        }
+
+        return Response.ok().entity(version.getMetricsByPlatform().get(platform)).build();
```

Comment:
If the metrics don't exist for this version/platform, should we return 404 or an empty response? I'm not sure, but if empty, then we should use no content, i.e., `Response.noContent().build();`, which will set the correct http status code.

Or does `version.getMetricsByPlatform().get()` always return a non-empty value?

Reply:
Following Kathy's comment, I changed the endpoint so it returns a map of all platforms instead. This way, we won't have to do individual requests for each platform